libsolidity/Parser.cpp

```diff
@@ -263,6 +266,39 @@ ASTPointer<StructDefinition> Parser::parseStructDefinition()
 	return nodeFactory.createNode<StructDefinition>(name, members);
 }
 
+ASTPointer<EnumValue> Parser::parseEnumDeclaration()
+{
+	ASTNodeFactory nodeFactory(*this);
+	ASTPointer<ASTString> name = expectIdentifierToken();
+	nodeFactory.markEndPosition();
+	return nodeFactory.createNode<EnumValue>(name);
+}
+
+ASTPointer<EnumDefinition> Parser::parseEnumDefinition()
+{
+	ASTNodeFactory nodeFactory(*this);
+	expectToken(Token::Enum);
+	ASTPointer<ASTString> name = expectIdentifierToken();
+	vector<ASTPointer<EnumValue>> members;
+	expectToken(Token::LBrace);
+
+	while (m_scanner->getCurrentToken() != Token::RBrace)
+	{
+		members.push_back(parseEnumDeclaration());
+		if (m_scanner->getCurrentToken() == Token::RBrace)
+			break;
+		expectToken(Token::Comma);
+		if (m_scanner->getCurrentToken() != Token::Identifier) {
+			BOOST_THROW_EXCEPTION(createParserError("Expected Identifier after ','"));
+		}
+	}
+
+	nodeFactory.markEndPosition();
+	expectToken(Token::RBrace);
+	expectToken(Token::Semicolon);
```

Comment:
Why the semicolon? It is not required by any other element that requires an explicit `}`.